ext/RastersStatsBaseExt/sample.jl

```diff
@@ -0,0 +1,85 @@
+Rasters.sample(x::RA.RasterStackOrArray, n::Integer; kw...) = Rasters.sample(Random.GLOBAL_RNG, x, n; kw...)
+@inline function Rasters.sample(
+    rng::Random.AbstractRNG, x::RA.RasterStackOrArray, n::Integer; 
+    geometry=(X,Y),
+    index = false, 
+    names=RA._names(x), 
+    name=names, 
+    skipmissing=false,
+    replace=true, 
+    ordered=false, 
+    weights=nothing, 
+    weightstype::Type{<:StatsBase.AbstractWeights}=StatsBase.Weights
+)
+    na = DD._astuple(name)
+    geometry, geometrytype, dims = _geometrytype(x, geometry)
+ #   x = x isa RA.AbstractRasterStack ? x[na] : x
+    _sample(rng, x, n;
+        dims=DD.dims(x, RA.DEFAULT_POINT_ORDER),
+        names=NamedTuple{na}(na),
+        # These keywords are converted to _True/_False for type stability later on
+        # The @inline above helps constant propagation of the Bools
+        geometry=geometry,
+        geometrytype=geometrytype,  
+        index=_booltype(index), 
+        skipmissing=_booltype(skipmissing), 
+        weights,
+        replace, 
+        ordered,
+        weightstype
+    )
+
+end
+function _sample(
+    rng, x, n; 
+    dims, names::NamedTuple{K}, geometry, geometrytype, index, skipmissing, weights, replace, ordered, weightstype
+) where K
+    indices = sample_indices(rng, x, n, skipmissing, weights, replace, ordered, weightstype)
+    points = DimPoints(dims)
+    T = RA._rowtype(x, geometrytype; geometry, index, skipmissing, skipinvalid = _True(), names)
+    x2 = x isa AbstractRasterStack ? x[K] : RasterStack(NamedTuple{K}((x,)))
+    _getindices(T, x2, points, indices)
+end
+
+_getindices(::Type{T}, x, points, indices) where T = 
+    broadcast(I -> _getindex(T, x, points, I), indices)
+
+_getindex(::Type{T}, x::AbstractRasterStack{<:Any, NT}, points, idx) where {T, NT} = 
+    RA._maybe_add_fields(T, NT(x[RA.commondims(idx, x)]), points[RA.commondims(idx, points)], val(idx))
```

Comment:
It's weird this NT trick is needed, maybe DD stack getindex stability is broken on 1.11?

Looks like something needs to be forced somewhere

Reply:
I know, nothing else worked. Even `AbstractRasterStack{K}` and then `NamedTuple{K}` wasn't type stable.

Reply:
No that makes sense, it's the eltype that's not propagating.

We might need some more `Base.assume_effects` in DD indexing